pkg/scheduler/schedule_one.go

```diff
@@ -810,7 +811,49 @@ func getAttemptsLabel(p *framework.QueuedPodInfo) string {
 // handleSchedulingFailure records an event for the pod that indicates the
 // pod has failed to schedule. Also, update the pod condition and nominated node name if set.
 func (sched *Scheduler) handleSchedulingFailure(ctx context.Context, fwk framework.Framework, podInfo *framework.QueuedPodInfo, err error, reason string, nominatingInfo *framework.NominatingInfo) {
-	sched.Error(podInfo, err)
+	pod := podInfo.Pod
+	if err == ErrNoNodesAvailable {
```

Comment:
this is moved from the other function, but lots of indentation in this segment of the code, perhaps we can try to improve it as a followup.